src/hotspot/os/posix/signals_posix.cpp

```diff
@@ -777,64 +823,48 @@ static void check_signal_handler(int sig) {
 
   os_sigaction(sig, (struct sigaction*)NULL, &act);
 
-  // See comment for SA_RESTORER_FLAG_MASK
-  LINUX_ONLY(act.sa_flags &= SA_RESTORER_FLAG_MASK;)
-
-  address thisHandler = get_signal_handler(&act);
-
-  switch (sig) {
-  case SIGSEGV:
-  case SIGBUS:
-  case SIGFPE:
-  case SIGPIPE:
-  case SIGILL:
-  case SIGXFSZ:
-    jvmHandler = CAST_FROM_FN_PTR(address, (sa_sigaction_t)javaSignalHandler);
-    break;
-
-  case SHUTDOWN1_SIGNAL:
-  case SHUTDOWN2_SIGNAL:
-  case SHUTDOWN3_SIGNAL:
-  case BREAK_SIGNAL:
-    jvmHandler = (address)os::user_handler();
-    break;
-
-  default:
-    if (sig == PosixSignals::SR_signum) {
-      jvmHandler = CAST_FROM_FN_PTR(address, (sa_sigaction_t)SR_handler);
-    } else {
-      return;
-    }
-    break;
-  }
+  // Compare both sigaction structures (intelligently; only the members we care about).
 
-  if (thisHandler != jvmHandler) {
+  // Check handler
+  address this_handler = get_signal_handler(&act);
+  address expected_handler = get_signal_handler(expected_act);
+  if (this_handler != expected_handler) {
+    mismatch = true;
     tty->print("Warning: %s handler ", os::exception_name(sig, buf, O_BUFLEN));
     tty->print_raw("expected:");
-    print_signal_handler_name(tty, jvmHandler, buf, O_BUFLEN);
+    print_signal_handler_name(tty, expected_handler, buf, O_BUFLEN);
     tty->print_raw("  found:");
-    print_signal_handler_name(tty, thisHandler, buf, O_BUFLEN);
-    // No need to check this sig any longer
-    sigaddset(&check_signal_done, sig);
+    print_signal_handler_name(tty, this_handler, buf, O_BUFLEN);
+    tty->cr();
     // Running under non-interactive shell, SHUTDOWN2_SIGNAL will be reassigned SIG_IGN
     if (sig == SHUTDOWN2_SIGNAL && !isatty(fileno(stdin))) {
       tty->print_cr("Running in non-interactive shell, %s handler is replaced by shell",
-                    os::exception_name(sig, buf, O_BUFLEN));
+                    os::exception_name(sig, buf, O_BUFLEN));    // When comparing, ignore the SA_RESTORER flag on Linux
```

Comment:
What does this comment mean in this context ???

Reply:
Nothing. This, like the "flags" comment before, must have been accidental paste errors. I went through the patch again, but I hope this was the last one.